core/src/test/java/org/elasticsearch/test/rest/client/RestClient.java

```diff
@@ -220,14 +245,52 @@ private RestApi restApi(String apiName) {
         return restApi;
     }
 
+    protected HttpRequestBuilder httpRequestBuilder(InetSocketAddress address) {
+        return new HttpRequestBuilder(httpClient)
+                .addHeaders(headers)
+                .protocol(protocol)
+                .host(address.getHostName()).port(address.getPort());
+    }
+
     protected HttpRequestBuilder httpRequestBuilder() {
         //the address used is randomized between the available ones
         InetSocketAddress address = RandomizedTest.randomFrom(addresses);
-        return new HttpRequestBuilder(httpClient).addHeaders(headers).host(address.getHostName()).port(address.getPort());
+        return httpRequestBuilder(address);
     }
 
-    protected CloseableHttpClient createHttpClient() {
-        return HttpClients.createMinimal(new PoolingHttpClientConnectionManager(15, TimeUnit.SECONDS));
+    protected CloseableHttpClient createHttpClient(Settings settings) throws IOException {
+        SSLConnectionSocketFactory sslsf;
+        String keystorePath = settings.get(TRUSTSTORE_PATH);
+        if (keystorePath != null) {
+            final String keystorePass = settings.get(TRUSTSTORE_PASSWORD);
+            if (keystorePass == null) {
+                throw new IllegalStateException(TRUSTSTORE_PATH + " is provided but not " + TRUSTSTORE_PASSWORD);
+            }
+            Path path = PathUtils.get(keystorePath);
+            if (!Files.exists(path)) {
+                throw new IllegalStateException(TRUSTSTORE_PATH + " is set but points to a non-existing file");
+            }
+            try {
+                KeyStore keyStore = KeyStore.getInstance("jks");
+                try (InputStream is = Files.newInputStream(path)) {
+                    keyStore.load(is, keystorePass.toCharArray());
+                }
+                SSLContext sslcontext = SSLContexts.custom()
+                        .loadTrustMaterial(keyStore, new TrustSelfSignedStrategy())
```

Comment:
can we get away with passing `null` instead of `TrustSelfSignedStrategy`? I'd prefer to have the certificates we trust only stored in the truststore